### ISSUE TITLE
Expressions that start with / are always printed in echo and list mode (maybe more).

### ISSUE THREAD

**Issue by externl** [fisherman/z#46]
```
❯ z /adsfasdfsadfsadf
cd: The directory '/adsfasdfsadfsadf' does not exist

❯ z -e /asdfasdfsadf
/asdfasdfsadf
```

I wouldn't expect anything to be printed for the second call. Also happens with `--list`. 

I tested with https://github.com/rupa/z, it prints nothing for the `z -e` case.

**Comment by jethrokuan** [comment]
I'm using this special case where expressions that start with `/` bypass `z`.

This is because you can use fish's autocomplete to generate paths that start with `/`. In that scenario, it doesn't need to go through `z` anymore.

Is this causing trouble for you? I use this quite often, and I think it makes sense.

**Comment by externl** [comment]
No trouble, I just noticed it when testing something else and noticed it's not how the `sh` version behaves. 

I don't really follow the use case though, the path I tested was "hardcoded" and do not plan on autocompleting.  Maybe I'm missing something.

**Comment by jethrokuan** [comment]
Suppose you have directories `foo` and `foobar` in the `$Z_DATA` file. Typing `z fo` and `<TAB>` wil show you the completions for the directories, specifically their absolute paths. You can then choose one to tab complete to, and the command will bypass `z` and `cd` into the directory.

This allows you to choose which folder to `z` into, since the completions are obtained by parsing the `$Z_DATA` file.

**Comment by externl** [comment]
Okay, but how does that relate to expressions that start with `/` and echo/list mode.

**Comment by jethrokuan** [comment]
That's precisely how echo works. It echoes the directory z will try to `cd` into. I can add another conditional to fix the list case.

**Comment by externl** [comment]
Okay, but I still don't follow, considering the two examples in my original comment. There is no completion (let's say they were scripted), why would the one starting with `/` always print?

**Comment by jethrokuan** [comment]
In the second case, the awk script `z` calls is not invoked at all, so what should the return result be?

**Comment by externl** [comment]
I think it should at least check that the directory exists.

**Comment by jethrokuan** [comment]
I'm inclined to just let the awk script run its course, don't think it'll be that much of a delay.

> I think it should at least check that the directory exists.

By this logic, before z tries to cd into the directory, it should check that the directory exists too. `Z_DATA` can contain directories that no longer exist.

**Comment by jethrokuan** [comment]
Should be fixed by #51

**Comment by externl** [comment]
Great, thanks.
